framework/src/samplers/Sampler.C

```diff
@@ -12,103 +12,76 @@
 /*            See COPYRIGHT for full restrictions               */
 /****************************************************************/
 
+// MOOSE includes
 #include "Sampler.h"
+#include "MooseRandom.h"
 #include "Distribution.h"
-#include <limits>
 
 template <>
 InputParameters
 validParams<Sampler>()
 {
   InputParameters params = validParams<MooseObject>();
-  params += validParams<RandomInterface>();
-  params.addParam<bool>("reseed", false, "Reseed for each new sample if this value is true");
-  params.addRequiredParam<std::vector<std::string>>(
-      "perturb_parameters", "The names of the parameters that you want to perturb");
+  params += validParams<SetupInterface>();
+  params += validParams<DistributionInterface>();
+
+  params.addClassDescription("A base class for distribution sampling.");
+  params.set<std::vector<OutputName>>("outputs") = {"none"};
   params.addRequiredParam<std::vector<DistributionName>>(
-      "distributions",
-      "The names of distributions that you want to use to perturb the given parameters");
+      "distributions", "The names of distributions that you want to sample.");
+  params.addParam<std::vector<unsigned int>>("seeds",
+                                             std::vector<unsigned int>(1, 19800624),
```

Comment:
Actually, it's unclear to me how many seeds a user would need to set. Is this documented anywhere? Why not just one seed? Then generate the rest of the seeds from the first? This is one of those things that can really screw a user if they use it incorrectly. I'd rather handle it in the design of the object.